src/main/java/teammates/ui/pagedata/FeedbackSubmissionEditPageData.java

```diff
@@ -249,26 +261,112 @@ private FeedbackSubmissionEditQuestion createQuestion(FeedbackQuestionAttributes
+                    questionAttributes.recipientType);
+
+            Map<String, String> commentGiverEmailToNameTable = bundle.getRoster().getEmailToNameTableFromRoster();
+            if (questionAttributes.getQuestionDetails().isFeedbackParticipantCommentsOnResponsesAllowed()) {
+
+                FeedbackResponseCommentRow responseCommentRow = getResponseCommentRowForResponse(questionAttributes,
+                        existingResponse.getId(), commentGiverName, commentRecipientName, commentGiverEmailToNameTable);
+
+                FeedbackResponseCommentRow frcForAdding = buildFeedbackResponseCommentAddFormTemplate(
+                        questionAttributes, existingResponse.getId(), commentGiverName,
+                        commentRecipientName, sessionTimeZone, true);
+
+                responses.add(new FeedbackSubmissionEditResponse(responseIndx,
+                        true, recipientOptionsForQuestion, submissionFormHtml,
+                        existingResponse.getId(), responseCommentRow, frcForAdding));
+
+                responseSubmittedRecipient.add(commentRecipientName);
+            } else {
+                responses.add(new FeedbackSubmissionEditResponse(responseIndx,
+                        true, recipientOptionsForQuestion,
+                        submissionFormHtml, existingResponse.getId(), null, null));
+            }
             responseIndx++;
         }
-
+        int recipientIndxForUnsubmittedResponse = 0;
         while (responseIndx < numOfResponseBoxes) {
             List<String> recipientOptionsForQuestion = getRecipientOptionsForQuestion(questionAttributes.getId(), null);
-            String submissionFormHtml = questionAttributes.getQuestionDetails()
-                                            .getQuestionWithoutExistingResponseSubmissionFormHtml(
-                                                isSessionOpenForSubmission, qnIndx, responseIndx,
-                                                questionAttributes.courseId, numOfResponseBoxes, student);
 
-            responses.add(new FeedbackSubmissionEditResponse(responseIndx, false, recipientOptionsForQuestion,
-                                                             submissionFormHtml, ""));
+            String submissionFormHtml = questionAttributes.getQuestionDetails()
+                    .getQuestionWithoutExistingResponseSubmissionFormHtml(
+                            isSessionOpenForSubmission, qnIndx, responseIndx,
+                            questionAttributes.courseId, numOfResponseBoxes, student);
+
+            if (questionAttributes.getQuestionDetails().isFeedbackParticipantCommentsOnResponsesAllowed()) {
+                List<String> recipientListForUnsubmittedResponse = getRecipientList(responseSubmittedRecipient,
+                        bundle.getSortedRecipientList(questionAttributes.getId()));
+                commentRecipientName = recipientListForUnsubmittedResponse.get(recipientIndxForUnsubmittedResponse);
```

Comment:
What's the different between getting recipient name from here and from roster?

Reply:
Can use `getCommentRecipientNameFromEmail `?

Reply:
Here we are filtering recipients name for whom response wasn't submitted from all recipients' names. Then from that list we make comment add forms. We can't use `getCommentRecipientNameFromEmail` because for that we need email of the recipient.

Reply:
Do we really need the recipient and giver name displayed as it is already displayed in the response form?

Reply:
Although I don't think it is really needed, I thought to make it consistent with comment add forms for instructors. Plus it gives clarity on who you're giving comment to. What do you think?

Reply:
My concern is about the design. How do you know that the current recipient is `recipientIndxForUnsubmittedResponse` in `recipientListForUnsubmittedResponse`? From your implementation, I guess because we iterating the options in a certain order and therefore if we sort the recipient and get by the same order, we will get the recipient. However, do you see the coupling here? I believe the whole function is designed to expose the recipient name.

Reply:
> From your implementation, I guess because we iterating the options in a certain order and therefore if we sort the recipient and get by the same order, we will get the recipient.

That's right. How else shall we know who is the recipient. Empty response exists in form of `String submissionFormHtml`. If we want to show recipient name then I guess we will have to use this coupling?

Reply:
> How else shall we know who is the recipient.

Unfortunately, we do this by javascript for displaying recipient name for response.

> If we want to show recipient name then I guess we will have to use this coupling?

I don't think introduce this coupling is a good option. My suggestions is to no show recipient name. But let's have @damithc's input for this.

If we have giver/recipient name:
![screen shot 2018-07-07 at 2 14 11 pm](https://user-images.githubusercontent.com/13977235/42407493-21736874-81f0-11e8-9638-1ae1f120efb8.png)

If not:
![screen shot 2018-07-07 at 2 14 43 pm](https://user-images.githubusercontent.com/13977235/42407503-49a737f8-81f0-11e8-8ab5-004d4179c82a.png)

Actually, no matter what implementation we choose, we shall still add something like 'You comment about your feedback' or students won't know what the input box is.

Reply:
Alright. Sounds good to me.